src/pool/srv_target.c

```diff
@@ -1945,39 +1953,23 @@ ds_pool_tgt_map_update(struct ds_pool *pool, struct pool_buf *buf,
 	return rc;
 }
 
-static inline void
-dps_to_dpsv6(struct daos_pool_space *ps, struct daos_pool_space_v6 *ps6)
-{
-	int i;
-
-	ps6->ps_ntargets = ps->ps_ntargets;
-	for (i = DAOS_MEDIA_SCM; i < DAOS_MEDIA_MAX; i++) {
-		ps6->ps_space.s_total[i] = ps->ps_space.s_total[i];
-		ps6->ps_space.s_free[i]  = ps->ps_space.s_free[i];
-
-		ps6->ps_free_max[i]  = ps->ps_free_max[i];
-		ps6->ps_free_min[i]  = ps->ps_free_min[i];
-		ps6->ps_free_mean[i] = ps->ps_free_mean[i];
-	}
-}
-
 static void
 pool_tgt_query_handler(crt_rpc_t *rpc, int handler_version)
 {
 	struct pool_tgt_query_in	*in = crt_req_get(rpc);
 	struct pool_tgt_query_out	*out = crt_reply_get(rpc);
-	struct daos_pool_space           out_space = {0}, *ps;
 	struct ds_pool			*pool;
+	uint64_t			*mem_file_bytes;
 	int				 rc;
 
-	if (handler_version == 6)
-		ps = &out_space;
+	if (handler_version >= 7)
+		mem_file_bytes = &out->tqo_mem_file_bytes;
 	else
-		ps = &out->tqo_space;
+		mem_file_bytes = NULL;
```

Comment:
nit: makes sense to just initialise pointer to null when declaring and remove this else branch